tests/instancetype/inference.go

```diff
@@ -0,0 +1,350 @@
+			Name: inferFromVolumeName,
+			VolumeSource: virtv1.VolumeSource{
+				DataVolume: &virtv1.DataVolumeSource{
+					Name: sourceDV.Name,
+				},
+			},
+		}}
+		createAndValidateVirtualMachine()
+	})
+
+	DescribeTable("should infer defaults from DataVolumeTemplates", func(generateDataVolumeTemplatesFunc func() []virtv1.DataVolumeTemplateSpec) {
+		vm.Spec.DataVolumeTemplates = generateDataVolumeTemplatesFunc()
+		vm.Spec.Template.Spec.Volumes = generateVolumesForDataVolumeTemplates()
+		createAndValidateVirtualMachine()
+	},
+		Entry("and DataVolumeSourcePVC",
+			func() []virtv1.DataVolumeTemplateSpec {
+				dv := libdv.NewDataVolume(
+					libdv.WithNamespace(namespace),
+					libdv.WithForceBindAnnotation(),
+					libdv.WithPVCSource(sourceDV.Namespace, sourceDV.Name),
+					libdv.WithStorage(libdv.StorageWithAccessMode(k8sv1.ReadWriteOnce), libdv.StorageWithVolumeSize("1Gi")),
+				)
+				return []virtv1.DataVolumeTemplateSpec{{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: dataVolumeTemplateName,
+					},
+					Spec: dv.Spec,
+				}}
+			},
+		),
+		Entry(", DataVolumeSourceRef and DataSource",
+			func() []virtv1.DataVolumeTemplateSpec {
+				By("Creating a DataSource")
+				// TODO - Replace with libds?
+				dataSource := &cdiv1beta1.DataSource{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "datasource-",
+						Namespace:    namespace,
+					},
+					Spec: cdiv1beta1.DataSourceSpec{
+						Source: cdiv1beta1.DataSourceSource{
+							PVC: &cdiv1beta1.DataVolumeSourcePVC{
+								Name:      sourceDV.Name,
+								Namespace: namespace,
+							},
+						},
+					},
+				}
+				dataSource, err := virtClient.CdiClient().CdiV1beta1().DataSources(namespace).Create(context.Background(), dataSource, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				dataVolume := libdv.NewDataVolume(
+					libdv.WithNamespace(namespace),
+					libdv.WithForceBindAnnotation(),
+					libdv.WithStorage(libdv.StorageWithAccessMode(k8sv1.ReadWriteOnce), libdv.StorageWithVolumeSize("1Gi")),
+					libdv.WithDataVolumeSourceRef("DataSource", namespace, dataSource.Name),
+				)
+
+				return generateDataVolumeTemplatesFromDataVolume(dataVolume)
+			},
+		),
+		Entry(", DataVolumeSourceRef and DataSource with labels",
+			func() []virtv1.DataVolumeTemplateSpec {
+				By("Creating a blank DV and PVC without labels")
+				blankDV := libdv.NewDataVolume(
+					libdv.WithNamespace(namespace),
+					libdv.WithForceBindAnnotation(),
+					libdv.WithBlankImageSource(),
+					libdv.WithStorage(libdv.StorageWithAccessMode(k8sv1.ReadWriteOnce), libdv.StorageWithVolumeSize("1Gi")),
+				)
+				blankDV, err := virtClient.CdiClient().CdiV1beta1().DataVolumes(namespace).Create(context.Background(), blankDV, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				libstorage.EventuallyDV(sourceDV, dvSuccessTimeoutSeconds, matcher.HaveSucceeded())
+
+				By("Creating a DataSource")
+				// TODO - Replace with libds?
```

Comment:
https://github.com/kubevirt/kubevirt/pull/15498 was my fup for this, feel free to take it over if you have time.